mongodbatlas/clusters.go

```diff
@@ -21,19 +21,27 @@ import (
 	"net/url"
 )
 
+type ChangeStatus string
+
+const (
+	ChangeStatusApplied ChangeStatus = "APPLIED"
+	ChangeStatusPending ChangeStatus = "PENDING"
+)
+
 const clustersPath = "groups/%s/clusters"
 
 // ClustersService is an interface for interfacing with the Clusters
 // endpoints of the MongoDB Atlas API.
 // See more: https://docs.atlas.mongodb.com/reference/api/clusters/
 type ClustersService interface {
-	List(context.Context, string, *ListOptions) ([]Cluster, *Response, error)
-	Get(context.Context, string, string) (*Cluster, *Response, error)
-	Create(context.Context, string, *Cluster) (*Cluster, *Response, error)
-	Update(context.Context, string, string, *Cluster) (*Cluster, *Response, error)
-	Delete(context.Context, string, string) (*Response, error)
-	UpdateProcessArgs(context.Context, string, string, *ProcessArgs) (*ProcessArgs, *Response, error)
-	GetProcessArgs(context.Context, string, string) (*ProcessArgs, *Response, error)
+	List(ctx context.Context, groupID string, options *ListOptions) ([]Cluster, *Response, error)
+	Get(ctx context.Context, groupID, clusterName string) (*Cluster, *Response, error)
+	Create(ctx context.Context, groupID string, cluster *Cluster) (*Cluster, *Response, error)
+	Update(ctx context.Context, groupID, clusterName string, cluster *Cluster) (*Cluster, *Response, error)
+	Delete(ctx context.Context, groupID, clusterName string) (*Response, error)
+	UpdateProcessArgs(ctx context.Context, groupID, clusterName string, args *ProcessArgs) (*ProcessArgs, *Response, error)
+	GetProcessArgs(ctx context.Context, groupID, clusterName string) (*ProcessArgs, *Response, error)
+	Status(ctx context.Context, groupID, clusterName string) (ClusterStatus, *Response, error)
```

Comment:
changed the signatures to include the names of parameters - this allows to distinguish two `string` parameters easier while working with the library. Found myself going to implementations regularly to make sure what parameters mean.
This is not critical and I can revert this in case more concise signatures are preferable!